grails-app/views/stockMovement/addComment.gsp

```diff
@@ -29,9 +30,25 @@
 			<g:render template="summary" model="[stockMovement:stockMovement]" />
 			<div class="box">
 				<h2><warehouse:message code="default.add.label" args="[entityName]" /></h2>
-				<g:form action="${comment?.id ? 'updateComment' : 'saveComment'}">
-					<g:hiddenField name="id" value="${comment?.id}" />
+
+				<g:set var="formAction"  value="saveComment" />
+				<g:set var="isRequestRejected" value="${approvalStatus == StockMovementStatusCode.REJECTED}" />
+				<g:if test="${comment?.id}">
+					<g:set var="formAction"  value="updateComment" />
+				</g:if>
+				<g:elseif test="${isRequestRejected}">
```

Comment:
since you don't have more statements below, `g:else` should be enough here.

Reply:
What if Dariusz wants to catch if request is rejected here, but not approved without comment?

Reply:
btw I feel like `isRejected` should be enough

Reply:
yes, you both are right, I have not realized that you have some statement inside the elseif.